setzer/document/preview/preview_viewgtk.py

```diff
@@ -48,6 +48,7 @@ def __init__(self):
         self.drawing_area = Gtk.DrawingArea()
         self.drawing_area.add_events(Gdk.EventMask.BUTTON_PRESS_MASK)
         self.drawing_area.add_events(Gdk.EventMask.BUTTON_RELEASE_MASK)
+        self.drawing_area.add_events(Gdk.EventMask.BUTTON2_MOTION_MASK)
```

Comment:
> I tried using existing `on_hover`, it wasn't working and checking with GTK inspector, signal is blocked when button is pressed. I'm probably unaware of something here.

https://docs.gtk.org/gdk3/flags.EventMask.html mentions:

> Since GTK 3.8, motion events are already compressed by default, independent of this mechanism. This compression can be disabled with gdk_window_set_event_compression(). See the documentation of that function for details.

https://docs.gtk.org/gdk3/method.Window.set_event_compression.html

Reply:
The compression is not related to this. You should be able to use the drag-signals, like "drag-update", "drag-end", ...

Reply:
instead of tracking the pressed buttons you may just need the panning start pos. but you should set that to none when no drag is in progress.